### PR TITLE
`tags-path` and `branches-path` JSON schema update

### PR DESCRIPTION
**Pull request by xylesoft**
Implemented JSON schema update to validate the `tags-path` and `branches-path` configuration clauses, also enabled the possibility of using arrays for both path configurations.

This Pull Request relates to: https://github.com/composer/satis/issues/312 and https://github.com/composer/composer/pull/5142

### CODE REVIEW

File: res/satis-schema.json [hunk 1]
```diff
@@ -78,7 +78,44 @@
...
+            "additionalProperties": true,
+            "items": {
+                "type": "object",
+                "additionalProperties": false,
+                "required": ["type","url"],
+                "properties": {
+                    "type" : {
+                        "type": "string",
+                        "description": "Version control system type",
+                        "enum": ["vcs","svn","git","hg"]
```
Comment (stof) [inline res/satis-schema.json]: that's wrong. There are more type of repositories

Comment (stof) [inline res/satis-schema.json]: missing ones: ``composer`` and ``package`` at least.

**Comment by stof** [comment]
``repositories`` should not have a strict validation in the Satis schema if it does not have it in the composer schema (as this config is about configuring composer itself)

**Comment by stof** [comment]
The current PR will break the usage of Satis for many cases, as it misses lots of config

**Comment by xylesoft** [comment]
Thank you for the feedback @stof, could you please give more some details of the missing areas regarding the config?

Regarding the `repositories`, which repositories are missing (besides composer and package)? I'm in the understanding that the schema should be an outline to the implementations permissible repositories.

**Comment by alcohol** [comment]
https://getcomposer.org/doc/05-repositories.md#types

Not sure how completely up to date the documentation is, but it should contain everything.

File: res/satis-schema.json (hunk 1 above)
Comment (alcohol) [inline res/satis-schema.json]: `artifact` also

**Comment by alcohol** [comment]
Not sure what to do with this since your PR is also still open @ Composer.

/cc @Seldaek can you provide some feedback? See also referenced issue #5142

**Pull request closed without merging**